### ISSUE TITLE
2.1.0 installs multiple versions of React

### ISSUE THREAD

**Issue by islemaster** [casesandberg/react-color#149]
```
> npm install
...
> npm ls react
react-color@2.1.0 /home/brad/Projects/casesandberg/react-color
├── react@15.0.2 
├─┬ react-context@0.0.3
│ └── react@0.13.3 
└─┬ reactcss@0.4.5
  └── react@0.14.8 
```

This is a minor issue, but one that we've found can bloat our build size if we're not careful.

Repros on a fresh pull of master at f603122813b82eedfa45f04c86fb7f695c2e9b6f although interactions with external dependencies also influence the repro.

I'm less concerned about [`react-context`](https://github.com/casesandberg/react-context) because it's listed in `devDependencies` and therefore only affects development of this package.  However, [`reactcss`](https://github.com/casesandberg/reactcss) is listed in `dependencies` and therefore gets installed by consumers of `react-color` - thus, consumers of `react-color` are also installing multiple versions of React at the moment.

The resolution is not a change in this repo, but it seems to impact react-color the most which is why I'm reporting it here.  reactcss/master supports React 15 as of https://github.com/casesandberg/reactcss/pull/52, but a new version of the package has not yet been released.  Once a new minor/patch version of reactcss is on npm, thanks to the `^0.4.3` rule consumers of `react-color` will no longer be installing two versions of React.

**Comment by islemaster** [comment]
For anyone interested, [here's our temporary workaround by depending on our own fork](https://github.com/code-dot-org/code-dot-org/pull/8274) until updated `reactcss` gets published.

**Comment by jonjaques** [comment]
I was able to solve this issue by switching the OR declaration around in both `react-color` & `reactcss`. It's odd, because `react-color@2.1.0` depends on `reactcss@^0.4.3` which resolves to `0.4.5`.

`reactcss@0.4.5` declares react deps in the same way as `react-color` does, with `0.14` coming first. However, when npm install does it's magic; it ends up changing the dep for reactcss to *just* `^0.14.0`

This seems to work though...
```
"react": "^15.0.0 || ^0.14.0",
"react-dom": "^15.0.0 || ^0.14.0"
```

Anyways, I'm tired and sick of this dependency hell, and what seems like an NPM bug.  😪

**Comment by hmeerlo** [comment]
Any news on this? reactcss has moved along and is on version 1.0.4 already supporting 0.14 and 15.1 as peerDependencies.

**Comment by islemaster** [comment]
It looks like this is still an issue, as reported a few days ago in https://github.com/casesandberg/react-color/issues/199.  I just tried installing react-color 2.2.2 and I end up with multiple versions of React:

```
╰○ npm ls react
(my app)
├─┬ react-color@2.2.2
│ ├── react@15.3.1 
│ └─┬ reactcss@0.4.6
│   └── react@0.14.8 
```

It looks like a fix is available in https://github.com/casesandberg/react-color/pull/201 though!

**Comment by casesandberg** [comment]
The fix is live `react-color@2.2.3`
